### PR TITLE
chore(pie-monorepo): DSW-000 auto updated lockfile and dangerjs check for missing lock file

### PR DESCRIPTION
**Pull request by jamieomaguire**
- Automatically commit the updated `yarn.lock` file if its a Version Packages PR
- Fail a danger check if the `yarn.lock` is missing after dependency updates if it is NOT a Version Packages PR

Example auto-commit:
Screenshot 2024-08-22 at 12 25 14

Example Danger message:
Screenshot 2024-08-22 at 12 25 52

### CODE REVIEW

File: packages/components/pie-webc-core/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "test:watch": "run -T vitest"
   },
   "dependencies": {
-    "lit": "3.1.3"
+    "lit": "3.2.0"
```
Comment (jamieomaguire) [inline packages/components/pie-webc-core/package.json:21]: Just a test upgrade to trigger the new CI step and danger check

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -56,8 +56,50 @@ jobs:
...
+              echo "lockfile_changed=true" >> $GITHUB_ENV
+            fi
+
+        # Commit and push changes if yarn.lock was updated
+        - name: Commit and push changes
+          if: env.lockfile_changed == 'true'
+          uses: stefanzweifel/git-auto-commit-action@v5
+          with:
+            commit_message: "chore: update yarn.lock"
+            commit_options: "--no-verify"
```
[the remark this answers is not in the document]
Comment (siggerzz) [reply .github/workflows/ci.yml:96]: woops 😅

File: dangerfile.js [hunk 3]
```diff
@@ -1,11 +1,31 @@
...
+const { execSync } = require('child_process');
+
 const { pr } = danger.github;
 const validChangesetCategories = ['Added', 'Changed', 'Removed', 'Fixed'];
 
 const isRenovatePR = pr.user.login === 'renovate[bot]';
 const isDependabotPR = pr.user.login === 'dependabot[bot]';
 
+// Only run the yarn lock check if the PR title is not "Version Packages"
+if (pr.title !== 'Version Packages') {
```
Comment (siggerzz) [inline dangerfile.js:12]:
```suggestion
const isDependabotPR = pr.user.login === 'dependabot[bot]';
const isPiebotPR = pr.user.login === 'pie-design-system-bot';

// Only run the yarn lock check if the PR title is not "Version Packages"
if (!isPiebotPR) {
```

**Review by siggerzz — DISMISSED** [review]
smol comments, but nice work! 🎉

**Review by siggerzz — DISMISSED** [review]

**Review by siggerzz — APPROVED** [review]

**Review by raoufswe — APPROVED** [review]

**Review by siggerzz — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -342,5 +384,5 @@ jobs:
       uses: re-actors/alls-green@release/v1
       with:
         # "allowed-skips" lists jobs that are optional but should not fail
-        allowed-skips: browser-tests-components, deploy-docs, deploy-storybook, browser-tests-docs
+        allowed-skips: browser-tests-components, deploy-docs, deploy-storybook, browser-tests-docs, build-ubuntu-node-20, unit-tests
```
Comment (jamieomaguire) [inline .github/workflows/ci.yml:387]: This PR highlighted a bug in the existing setup - this change doesn't trigger a build or unit tests which meant CI failed. Allowing these two steps to be skipped (but not fail) is the desired behaviour

**Review by raoufswe — APPROVED** [review]

**Pull request merged**